src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -121,15 +124,41 @@ public void createDirectoryEntry(
       String partitionKey = getCollectionPartitionKey(collectionId, testPath);
       String rowKey = encodePathAsAzureRowKey(testPath);
       TableEntity entity = FireStoreDirectoryEntry.toTableEntity(partitionKey, rowKey, dirToCreate);
-      logger.info("Upserting directory entry for {} in table {}", testPath, tableName);
       // For file ingest worker flights,
       // It's possible that another thread is trying to write the same directory entity at the same
       // time upsert rather than create so that it does not fail if it already exists
-      tableClient.upsertEntity(entity);
+      upsertEntityCheckStatus(tableClient, entity, testPath, tableName);
     }
     createEntityForPath(tableClient, collectionId, tableName, createEntry);
   }
 
+  public void upsertEntityCheckStatus(
+      TableClient tableClient, TableEntity entity, String path, String tableName) {
+    try {
+      Response<Void> response =
+          tableClient.upsertEntityWithResponse(
+              entity, TableEntityUpdateMode.MERGE, Duration.ofSeconds(20), null);
+      if (response.getStatusCode() >= 200 && response.getStatusCode() < 300) {
```

Comment:
I'm used to the google client `HttpResponse` which has a `isSuccessStatusCode` method that is exactly this line.

Reply:
Yeah, I didn't immediately find a more standard way to check for a passing status code. But, definitely open to suggestions because this isn't ideal.